ibis/expr/types/relations.py

```diff
@@ -2671,20 +2668,67 @@ def sql(self, query: str, dialect: str | None = None) -> ir.Table:
         Examples
         --------
         >>> import ibis
+        >>> from ibis import _
         >>> ibis.options.interactive = True
         >>> t = ibis.examples.penguins.fetch(table_name="penguins")
-        >>> expr = t.sql("SELECT island, mean(bill_length_mm) FROM penguins GROUP BY 1 ORDER BY 2 DESC")
+        >>> expr = t.sql(
+        ...     """
+        ...     SELECT island, mean(bill_length_mm) AS avg_bill_length
+        ...     FROM penguins
+        ...     GROUP BY 1
+        ...     ORDER BY 2 DESC
+        ...     """
+        ... )
         >>> expr
-        ┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┓
-        ┃ island    ┃ mean(bill_length_mm) ┃
-        ┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━┩
-        │ string    │ float64              │
-        ├───────────┼──────────────────────┤
-        │ Biscoe    │            45.257485 │
-        │ Dream     │            44.167742 │
-        │ Torgersen │            38.950980 │
-        └───────────┴──────────────────────┘
-        """
+        ┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
+        ┃ island    ┃ avg_bill_length ┃
+        ┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
+        │ string    │ float64         │
+        ├───────────┼─────────────────┤
+        │ Biscoe    │       45.257485 │
+        │ Dream     │       44.167742 │
+        │ Torgersen │       38.950980 │
+        └───────────┴─────────────────┘
+
+        Mix and match ibis expressions with SQL queries
+
+        >>> t = ibis.examples.penguins.fetch(table_name="penguins")
+        >>> expr = t.sql(
+        ...     """
+        ...     SELECT island, mean(bill_length_mm) AS avg_bill_length
+        ...     FROM penguins
+        ...     GROUP BY 1
+        ...     ORDER BY 2 DESC
+        ...     """
+        ... )
+        >>> expr = expr.mutate(
+        ...     island=_.island.lower(),
+        ...     avg_bill_length=_.avg_bill_length.round(1),
+        ... )
+        >>> expr
+        ┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
+        ┃ island    ┃ avg_bill_length ┃
+        ┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
+        │ string    │ float64         │
+        ├───────────┼─────────────────┤
+        │ biscoe    │            45.3 │
+        │ dream     │            44.2 │
+        │ torgersen │            39.0 │
+        └───────────┴─────────────────┘
+        >>> expr.alias("b").sql("SELECT * FROM b WHERE avg_bill_length > 40")
```

Comment:
Can you add a brief sentence above this line saying why the alias is necessary here? Something like "If you want to use `.sql` on an existing ibis expression, you first need to assign it an alias to use in the query. This can be done with the `alias` method".